app/Http/Controllers/PokerController.php

```diff
@@ -1,133 +1,133 @@
-<?php
+    <?php
 
-namespace App\Http\Controllers;
+    namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
-use Illuminate\Support\Facades\DB;
-use Illuminate\Support\Facades\Broadcast;
-use Illuminate\Support\Facades\Log;
+    use Illuminate\Http\Request;
+    use Illuminate\Support\Facades\DB;
+    use Illuminate\Support\Facades\Broadcast;
+    use Illuminate\Support\Facades\Log;
 
-class PokerController extends Controller
-{
+    class PokerController extends Controller
+    {
 
-    public function createRoom(Request $request){
+        public function createRoom(Request $request){
 
-        $room_id = $this->generateRandomString(); //todo make sure this is unique in db        
+            $room_id = $this->generateRandomString(); //todo make sure this is unique in db        
 
-        $now = date('Y-m-d H:i:s');
-        $id = DB::table('poker_rooms')->insertGetId([
-            'room_id' => $room_id,
-            'password' => $request->password,
-            'is_open' => true,
-            'created_at' => $now,
-            'updated_at' => $now
-        ]);
-
-        $one_week_ago = date("Y-m-d H:i:s", strtotime("-1 week"));
+            $now = date('Y-m-d H:i:s');
+            $id = DB::table('poker_rooms')->insertGetId([
+                'room_id' => $room_id,
+                'password' => $request->password,
+                'is_open' => true,
+                'created_at' => $now,
+                'updated_at' => $now
+            ]);
 
-        DB::table('poker_rooms')->where('created_at', '<=', $one_week_ago)->update([
-            'is_open' => 0
-        ]);
-        return response()->json(['success' => true, 'room_id' => $room_id]);
-    }
+            $one_week_ago = date("Y-m-d H:i:s", strtotime("-1 week"));
 
-    function getData(Request $request){
-        $room = DB::table('poker_rooms')->where('room_id', $request->room_id)->first();
-        $players = DB::table('poker_players')->where('room_id', $room->id)->get();
-        return response()->json(['room' => $room, 'players' => $players]);
-    }
-
-    function getRooms(){
-        $rooms = DB::table('poker_rooms')->get();
+            DB::table('poker_rooms')->where('created_at', '<=', $one_week_ago)->update([
```

Comment:
in lieu of having a worker function cleanup my database, I decided to attach this logic to the room creation function which is a function that is guaranteed to execute (since it's the first step when it comes to using this service). Knowing that I want to avoid worker functions, is this still an issue?

Reply:
I am not very familiar with worker functions, but is a lot of work? is there no "onPageLoad" function we can attach this to (or maybe we can attach it to the controller constructor) Is this createRoom function called when the page loads or when a user explicitly clicks on createRoom ? 

If its the latter, than it might cause a weird use case in which i see data, create a room, then dont see the old data anymore, making me think there was a bug

Reply:
Thats a good point the constructor is a good place for that function